### PR TITLE
force port to always be evaluated as number

### PR DESCRIPTION
**Pull request by kukulaka**
when config was generated, port was being parsed as string. MSSQL driver does not like this.

### CODE REVIEW

File: api/src/database/index.ts [hunk 1]
```diff
@@ -18,9 +18,15 @@ for (let [key, value] of Object.entries(env)) {
...
 
-	connectionConfig[camelCase(key)] = value;
+	if (key === 'port') {
+		connectionConfig[camelCase(key)] = Number(value);
+	} else {
+		connectionConfig[camelCase(key)] = value;
+	}
 }
 
+console.log(connectionConfig);
```
Comment (rijkvanzanten) [inline api/src/database/index.ts]: Sneaky console log!

File: api/src/database/index.ts [hunk 2]
```diff
@@ -18,9 +18,15 @@ for (let [key, value] of Object.entries(env)) {
 
 	key = key.slice(3); // remove `DB_`
 
-	connectionConfig[camelCase(key)] = value;
+	if (key === 'port') {
+		connectionConfig[camelCase(key)] = Number(value);
+	} else {
+		connectionConfig[camelCase(key)] = value;
+	}
```
Comment (rijkvanzanten) [inline api/src/database/index.ts:25]: I'm wondering if we should try this for any key in `/src/env.ts` instead. There's quite a couple "number" configurations (like rate limiter points) that would be convenient to convert ahead of time 🤔

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: api/src/database/index.ts (hunk 2 above)
Comment (kukulaka) [reply api/src/database/index.ts:25]: @rijkvanzanten that sounds like a good idea to me 💯

File: api/src/database/index.ts [hunk 3]
```diff
@@ -18,7 +19,11 @@ for (let [key, value] of Object.entries(env)) {
 
 	key = key.slice(3); // remove `DB_`
 
-	connectionConfig[camelCase(key)] = value;
+	if (isNumeric(value)) {
+		connectionConfig[camelCase(key)] = Number(value);
+	} else {
+		connectionConfig[camelCase(key)] = value;
+	}
```
Comment (rijkvanzanten) [inline api/src/database/index.ts:26]: We should do this in `src/env.ts` instead 🙂  (across all env vars)

Comment (mscbpi) [reply api/src/database/index.ts:26]: @rijkvanzanten  may you review https://github.com/directus/next/pull/623 ?

Comment (rijkvanzanten) [reply api/src/database/index.ts:26]: I may! (no need to tag me btw, I see everything coming in/going out of directus/* 😉 )

**Comment by rijkvanzanten** [comment]
Closed in favor of https://github.com/directus/next/pull/623

**Pull request closed without merging**